src/rtctools_heat_network/workflows/io/write_output.py

```diff
@@ -689,6 +743,156 @@ def _name_to_asset(name):
                     asset = _name_to_asset(p)
                     asset.delete(recursive=True)
 
+        # ------------------------------------------------------------------------------------------
+        # Important: This code below must be placed after the "Placement" code. Reason: it relies
+        # on unplaced assets being deleted.
+        # ------------------------------------------------------------------------------------------
+        # Write asset result profile data to database:
+        #  - The database contains a measurement (used esdl energy system id as the name for this),
+        #    which is a table of the profile results. The each time step is represented by a row of
+        #    data, and the columns are: datetime, asset name + "_" variable value
+        #  - At the same time also set the database attributes for each asset with a profile data
+        if self.write_result_db_profiles:
+            logger.info("Writing asset result profile data to influxDB")
+            results = self.extract_results()
+            variables_one_hydraulic_system = ["HeatIn.Q", "HeatIn.H", "Heat_flow"]
+            variables_two_hydraulic_system = [
+                "Primary.HeatIn.Q",
+                "Primary.HeatIn.H",
+                "Secondary.HeatIn.Q",
+                "Secondary.HeatIn.H",
+                "Heat_flow",
+            ]
+            profiles = ProfileManager()
+            profiles.profile_type = "DATETIME_LIST"
+            profiles.profile_header = ["datetime"]
+
+            for ii in range(len(self.times())):
+                data_row = [self.io.datetimes[ii]]
+
+                for asset_name in [
+                    *self.heat_network_components.get("source", []),
+                    *self.heat_network_components.get("demand", []),
+                    *self.heat_network_components.get("pipe", []),
+                    *self.heat_network_components.get("buffer", []),
+                    *self.heat_network_components.get("ates", []),
+                    *self.heat_network_components.get("heat_exchanger", []),
+                    *self.heat_network_components.get("heat_pump", []),
+                ]:
+                    try:
+                        # If the asset has been placed
+                        asset = _name_to_asset(asset_name)
+
+                        # Get index of outport which will be used to assign the profile data to
+                        index_outport = -1
+                        for ip in range(len(asset.port)):
+                            if asset.port[ip].name == "Out":
+                                if index_outport == -1:
+                                    index_outport = ip
```

Comment:
maybe just escape the loop after you found the first outport to avoid HEX and HP to go into the else statement

Reply:
Agreed. Now I am assigning the profile to the 1st outport.

Reply:
Done. Please check